src/fetchers/tarball-fetcher.js

```diff
@@ -178,6 +185,17 @@ export default class TarballFetcher extends BaseFetcher {
           this.reporter.warn(this.reporter.lang('retryOnInternalServerError'));
           await sleep(3000);
         } else {
+          const tarballMirrorPath = this.getTarballMirrorPath();
+          const tarballCachePath = this.getTarballCachePath();
+
+          if (tarballMirrorPath && (await fsUtil.exists(tarballMirrorPath))) {
+            await fsUtil.unlink(tarballMirrorPath);
```

Comment:
@arcanis just curious, why does the unlinking have to be synchronous here?

Reply:
One benefit I see is that uncaught exceptions will be logged to the console, and will possibly abort the execution in future Node versions. Apart from that, it shouldn't really matter, but if you do it I suggest you also add a comment to explain it's made on purpose, since it can also be seen as a typo.